### PR TITLE
Fix dumping logs in regional and multizonal GKE clusters

### PR DESCRIPTION
**Pull request by jkaniuk**
Fix dumping logs in regional and multizonal GKE clusters

/cc @mborsz 
/cc @wojtek-t 
/cc @mm4tt

### CODE REVIEW

File: kubetest/gke.go [hunk 1]
```diff
@@ -462,12 +458,20 @@ export KUBE_NODE_OS_DISTRIBUTION='%[3]s'
...
+	var lastError error
+	for zone, filters := range perZoneFilters {
+		err := control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf(gkeLogDumpTemplate,
+			g.project,
+			zone,
+			os.Getenv("NODE_OS_DISTRIBUTION"),
+			strings.Join(filters, " OR "),
+			dumpCmd)))
+		if err != nil {
+			log.Printf("error while dumping logs in %s zone: %v", zone, err)
```
Comment (wojtek-t) [inline kubetest/gke.go:470]: Let's not print and return the same error

File: kubetest/gke.go [hunk 2]
```diff
@@ -462,12 +458,20 @@ export KUBE_NODE_OS_DISTRIBUTION='%[3]s'
 		log.Printf("error while ensuring firewall rule: %v", err)
 	}
 
-	return control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf(gkeLogDumpTemplate,
-		g.project,
-		g.zone,
-		os.Getenv("NODE_OS_DISTRIBUTION"),
-		strings.Join(filters, " OR "),
-		dumpCmd)))
+	var lastError error
```
Comment (wojtek-t) [inline kubetest/gke.go:461]: Instead of last, can we contatenate all?

File: kubetest/gke.go (hunk 1 above)
Comment (jkaniuk) [reply kubetest/gke.go:470]: I wanted to print it, as this error message is actually discarded in the calling function:
https://github.com/kubernetes/test-infra/blob/master/kubetest/e2e.go#L122

Comment (jkaniuk) [reply kubetest/gke.go:470]: Nevermind. That call was only for cases where test failed during up. In other places error is handled.

Comment (jkaniuk) [reply kubetest/gke.go]: Done

File: kubetest/gke.go (hunk 2 above)
Comment (jkaniuk) [reply kubetest/gke.go]: Done

File: kubetest/gke.go [hunk 3]
```diff
@@ -462,12 +458,22 @@ export KUBE_NODE_OS_DISTRIBUTION='%[3]s'
...
+	var errorMessages []string
+	for zone, filters := range perZoneFilters {
+		err := control.FinishRunning(exec.Command("bash", "-c", fmt.Sprintf(gkeLogDumpTemplate,
+			g.project,
+			zone,
+			os.Getenv("NODE_OS_DISTRIBUTION"),
+			strings.Join(filters, " OR "),
+			dumpCmd)))
+		if err != nil {
+			errorMessages = append(errorMessages, fmt.Sprintf("%v", err))
```
Comment (wojtek-t) [inline kubetest/gke.go:470]: nit:
errorMessages = append(errorMessages, err.Error())

Comment (jkaniuk) [reply kubetest/gke.go]: Done

**Pull request merged**